File: docs/breez-sdk/snippets/go/lnurl_auth.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-spark-go/breez_sdk_spark"
)

func parseLnurlAuth(sdk *breez_sdk_spark.BreezSdk) {
	// ANCHOR: parse-lnurl-auth
	// LNURL-auth URL from a service
	// Can be in the form:
	// - lnurl1... (bech32 encoded)
	// - https://service.com/lnurl-auth?tag=login&k1=...
	lnurlAuthUrl := "lnurl1..."

	inputType, err := sdk.Parse(lnurlAuthUrl)
	if err == nil {
		if lnurlAuth, ok := inputType.(breez_sdk_spark.InputTypeLnurlAuth); ok {
			requestData := lnurlAuth.Field0
			log.Printf("Domain: %s", requestData.Domain)
			log.Printf("Action: %v", requestData.Action)

			// Show domain to user and ask for confirmation
			// This is important for security
		}
	}
	// ANCHOR_END: parse-lnurl-auth
}

func authenticate(sdk *breez_sdk_spark.BreezSdk, requestData breez_sdk_spark.LnurlAuthRequestDetails) {
	// ANCHOR: lnurl-auth
	// Perform LNURL authentication
	result, err := sdk.LnurlAuth(requestData)
	if err != nil {
		log.Printf("Authentication error: %v", err)
		return
	}

	switch v := result.(type) {
	case breez_sdk_spark.LnurlCallbackStatusOk:
		log.Println("Authentication successful")
	case breez_sdk_spark.LnurlCallbackStatusErrorStatus:
		log.Printf("Authentication failed: %s", v.ErrorDetails.Reason)
	}
	// ANCHOR_END: lnurl-auth
}

File: docs/breez-sdk/snippets/csharp/LnurlAuth.cs
using Breez.Sdk.Spark;

namespace BreezSdkSnippets;

public static class LnurlAuth
{
    public static async Task ParseLnurlAuth(BreezSdk sdk)
    {
        // ANCHOR: parse-lnurl-auth
        // LNURL-auth URL from a service
        // Can be in the form:
        // - lnurl1... (bech32 encoded)
        // - https://service.com/lnurl-auth?tag=login&k1=...
        var lnurlAuthUrl = "lnurl1...";

        var inputType = await sdk.Parse(lnurlAuthUrl);
        if (inputType is InputType.LnurlAuth lnurlAuth)
        {
            var requestData = lnurlAuth.v1;
            Console.WriteLine($"Domain: {requestData.domain}");
            Console.WriteLine($"Action: {requestData.action}");

            // Show domain to user and ask for confirmation
            // This is important for security
        }
        // ANCHOR_END: parse-lnurl-auth
    }

    public static async Task Authenticate(BreezSdk sdk, LnurlAuthRequestDetails requestData)
    {
        // ANCHOR: lnurl-auth
        // Perform LNURL authentication
        var result = await sdk.LnurlAuth(requestData);

        if (result is LnurlCallbackStatus.Ok)
        {
            Console.WriteLine("Authentication successful");
        }
        else if (result is LnurlCallbackStatus.ErrorStatus errorStatus)
        {
            Console.WriteLine($"Authentication failed: {errorStatus.errorDetails.reason}");
        }
        // ANCHOR_END: lnurl-auth
    }
}
